starknet_py/contract.py

```diff
@@ -194,45 +196,28 @@ async def call(
 
     async def invoke(
         self,
-        signature: Optional[Collection[int]] = None,
         max_fee: Optional[int] = None,
         auto_estimate: bool = False,
     ) -> InvokeResult:
         """
         Invokes a method.
 
-        :param signature: Signature to send
         :param max_fee: Max amount of Wei to be paid when executing transaction
         :param auto_estimate: Use automatic fee estimation, not recommend as it may lead to high costs
         :return: InvokeResult
         """
-        if auto_estimate and max_fee is not None:
+        if not isinstance(self._client, AccountClient):
             raise ValueError(
-                "Max_fee and auto_estimate are exclusive and cannot be provided at the same time."
-            )
-        if auto_estimate and self.max_fee is not None:
-            raise ValueError(
-                "Auto_estimate cannot be used if max_fee was provided when preparing a function call."
+                "Use the AccountClient while creating Contract instance to invoke the transaction"
```

Comment:
I think something like "Contract uses an account that can't invoke transactions. You need to use AccountClient for that." explains it better.